opencti-platform/opencti-graphql/src/migrations/1712225747215-convert_linked-to_related-to.js

```diff
@@ -0,0 +1,128 @@
+import { logApp } from '../config/conf';
+import { elDeleteByQueryForMigration, elList, elReindexByQueryForMigration, elUpdateByQueryForMigration } from '../database/engine';
+import { INDEX_STIX_CORE_RELATIONSHIPS, READ_DATA_INDICES, INDEX_STIX_META_RELATIONSHIPS, READ_INDEX_STIX_META_RELATIONSHIPS, READ_RELATIONSHIPS_INDICES } from '../database/utils';
+import { RELATION_RELATED_TO } from '../schema/stixCoreRelationship';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+
+export const up = async (next) => {
+  const context = executionContext('migration');
+  logApp.info('[MIGRATION] Update all linked-to refs to related-to rels');
+
+  const linkedToType = 'x_opencti_linked-to';
+  const relLinkedTo = `rel_${linkedToType}.internal_id`;
+  const relRelatedTo = `rel_${RELATION_RELATED_TO}.internal_id`;
+
+  // We get all current related-to relations: will be used to check for duplicates in linked-to to related-to migration
+  const relatedToRelations = await elList(context, SYSTEM_USER, READ_RELATIONSHIPS_INDICES, { types: [RELATION_RELATED_TO] });
+  const relatedToRelationsFromToArray = relatedToRelations.map((rel) => rel.fromId + rel.toId);
+
+  // First, we reindex all linked-to meta refs to core rel index, updating all linked-to refs to related-to rel in the process
+  // During the process, we check for duplicates with existing relatedTo rel: if on already exists, we skip this element for the reindexing
+  const reindexLinkedToToRelatedToSource = `
+    String fromId;
+    String toId;
+    ctx._source.entity_type = params.relToType;
+    ctx._source.relationship_type = params.relToType;
+    ctx._source.parent_types = params.relToParentTypes;
+    ctx._source.standard_id = ctx._source.standard_id.replace("relationship-meta--", "relationship--");
+    for(connection in ctx._source.connections) {
+      if(connection.role == params.linkedToType + '_from' && connection.containsKey('internal_id')) {fromId = connection.internal_id}
+      if(connection.role == params.linkedToType + '_to' && connection.containsKey('internal_id')) {toId = connection.internal_id}
+      connection.role = connection.role.replace(params.linkedToType, params.relToType);
+    }
+    if(params.relatedToRelationsFromToArray.contains(fromId+toId)) {ctx.op = 'noop'}
```

Comment:
what does it mean `ct.op = 'noop'` ?

Reply:
it's used to skip the reindex operation for this given document. In the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html), in the "Modify documents during reindexing, there is also a ct.op = 'delete' that is mentioned, but I'm not sure what the difference between the 2 are.